### PR TITLE
Feature/overview screen

### PR DESCRIPTION
**Pull request by Kapestian**
This pull request introduces the "Overview" screen, also known as "Around You". At the moment, The UI uses a simplified version of the backend to demonstrate it's capabilities. Some soldering work will be needed after the PR to join the multiple screens to this one (filter, viewProfile).

### New UI Components and Functionality:

* [`AroundYou.kt`](diffhunk://#diff-239df64079fc273bfbbaca6f5584cef7b7d832c06b62d2c706f26382d14b12cdR4-R34): Implemented the `AroundYouScreen` composable function, which displays a list of profiles or a message when no profiles are available. Added a floating action button and a top bar. [[1]](diffhunk://#diff-239df64079fc273bfbbaca6f5584cef7b7d832c06b62d2c706f26382d14b12cdR4-R34) [[2]](diffhunk://#diff-239df64079fc273bfbbaca6f5584cef7b7d832c06b62d2c706f26382d14b12cdL23-R51) [[3]](diffhunk://#diff-239df64079fc273bfbbaca6f5584cef7b7d832c06b62d2c706f26382d14b12cdR60-R99)
* [`ProfileCard.kt`](diffhunk://#diff-5e1483377625ec3afe9c4c8c0a3e7d38c13ffed12101a904cc34a52a70a12182R1-R66): Created the `ProfileCard` composable for displaying individual profiles.
* [`TopBar.kt`](diffhunk://#diff-f5f25d5e4d5f595fe5e31c4eef95be57db3452acc9ad28d86c7772c3803ba6b8R1-R35): Added a `TopBar` composable for the top navigation bar.

### Mock Data and ViewModel:

* [`MockProfileBackend.kt`](diffhunk://#diff-a3f5bd19f1f1311928af94ccadbe9e56e0cbbd094933f78fd3ff00eb8d8f6379R1-R128): Added `MockProfileViewModel` to provide mock profile data for testing and demonstration. This file will be modified/removed as the project advances.
* [`Profile.kt`](diffhunk://#diff-d7b68330078953d8ab16c69321f48aa1a58adf225f8334a5e38138385a3016d3R42-R43): Added a companion object to the `Profile` data class to support mock profile generation.

>[!NOTE]
>the last two composable are available in the `com/github/se/icebreakrr/ui/sections/shared` folder to be used in other screens

### Tests:

* [`AroundYouTest.kt`](diffhunk://#diff-a1e6284c2f712f779440c996db75a136ba136f8fef1543ece498997a3e245dd1R1-R46): Introduced tests for the "Around You" screen to verify the display of components and behavior when profiles are empty.

### Drawables:

* Added new drawable resources for icons used in the UI. [[1]](diffhunk://#diff-5521ef5a32cd6e3ee0e45c473d8f5b748573869fd9641faebd9e423103059000R1-R5) [[2]](diffhunk://#diff-1ca759ba02bc80fb5176c0b38f9c65297f251e24d148375b79bf60a93dba990fR1-R5) [[3]](diffhunk://#diff-8dd038c9df318e7a1d9f42ade3b0f4ebf88faa3e7968a5fa31bb57036c651e4aR1-R5) [[4]](diffhunk://#diff-f450327cf2d87b55bbac0a0c9bff5270f025aaeed9331d921094db5593ad125cR1-R5)

### CODE REVIEW

**Review by MaxCastelCast — APPROVED** [review]

File: app/src/main/java/com/github/se/icebreakrr/model/profile/Profile.kt [hunk 1]
```diff
@@ -39,6 +39,8 @@ data class Profile(
     }
     return age
   }
+
+  companion object
```
Comment (MaxCastelCast) [inline app/src/main/java/com/github/se/icebreakrr/model/profile/Profile.kt:43]: I don't really understand this part, it isn't very clear

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt [hunk 2]
```diff
@@ -29,11 +57,45 @@ fun AroundYouScreen(navigationActions: NavigationActions) {
...
+                    fontWeight = FontWeight.Bold,
+                    color = Color(0xFF575757))
+              }
+        }
+      },
+      floatingActionButton = {
+        val context = LocalContext.current
+        FloatingActionButton(
+            onClick = {
+              Toast.makeText(context, "Filter button clicked", Toast.LENGTH_SHORT).show()
```
Comment (MaxCastelCast) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt:93]: This filter Button is very similar from the one i have in Notification, i think we could use the same FAB for the two screens

**Review by MaxCastelCast — APPROVED** [review]
Looks good to me

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt (hunk 2 above)
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt:93]: As I've seen your version of the fab which is far prettier than mine, could you make it a component so we can later harmonize our screens?

File: app/src/main/java/com/github/se/icebreakrr/model/profile/Profile.kt (hunk 1 above)
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/model/profile/Profile.kt:43]: it is actually used in my mock file to create an extension. While your current companion object doesn't have any content, it's harmless and can be useful if you plan to add shared members or utility functions for the Profile class later.

**Pull request merged**